src/bin/electrs.rs

```diff
@@ -142,6 +145,25 @@ fn run_server(config: Arc<Config>) -> Result<()> {
     Ok(())
 }
 
+#[cfg_attr(feature = "otlp-tracing", tokio::main)]
+#[cfg_attr(not(feature = "otlp-tracing"), allow(dead_code))]
+async fn main_async() {
+    #[cfg(feature = "otlp-tracing")]
+    let _tracing_guard = otlp_trace::init_tracing("electrs");
+
+    let config = Arc::new(Config::from_args());
```

Comment:
To avoid repeating this code here and in `main()` and to avoid `allow(dead_code)`, I would restructure it like this:

```rust
fn main_() {
    let config = Arc::new(Config::from_args());
    // ... rest of body
}

#[cfg(not(feature = "otlp-tracing"))]
fn main() { main_() }

#[cfg(feature = "otlp-tracing")]
#[tokio::main]
async fn main() {
    let _tracing_guard = otlp_trace::init_tracing("electrs");
    main_()
}
```

Reply:
Good point!